pest/src/lib.rs

```diff
@@ -5,23 +5,38 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-extern crate futures;
+use std::fmt::Debug;
+use std::hash::Hash;
 
 mod error;
-mod inputs;
+mod inputs_private;
+mod iterators_private;
+mod macros;
 mod parser;
 mod parser_state;
-mod streams_private;
-pub mod tokens;
+mod token;
 
-/// A `mod` containing `Stream` implementations used in `Token` processing.
-pub mod streams {
-    pub use super::streams_private::{ExpandedStream, ParserStream, TokenDataFuture, TokenStream,
-                                     TailStream};
+/// A `mod` containing the `Input`-related constructs.
+pub mod inputs {
+    pub use super::inputs_private::{Input, Position, Span, StringInput};
 }
 
+/// A `mod` containing iterators and constructs to aid in parser output manipulation.
+pub mod iterators {
+    pub use super::iterators_private::{Pair, Pairs, TokenIterator};
+}
+
+/// A `trait` which parser rules must implement.
```

Comment:
Note that I added some documentation here to clarify *why* this trait works. Let me know if you have any feedback.